tests/test_maintel_setupmtcs.py

```diff
@@ -0,0 +1,127 @@
+
+# Clean up log to focus only on script
+logging.basicConfig(level=logging.DEBUG)
+logging.getLogger("asyncio").setLevel(logging.ERROR)
+logging.getLogger("lsst.ts.utils.tai").setLevel(logging.ERROR)
+logging.getLogger("MTCSMock").setLevel(logging.ERROR)
+
+
+class TestSetupMTCS(BaseScriptTestCase, unittest.IsolatedAsyncioTestCase):
+    async def basic_make_script(self, index):
+        self.script = SetupMTCS(index=index)
+        self.mtcs_mock = MTCSMock()
+        return self.script, self.mtcs_mock
+
+    async def test_configure_errors(self):
+        """Test error handling in the do_configure method."""
+        async with self.make_script():
+
+            # Check schema validation.
+            for bad_config in (
+                # The only *bad* case is that ccw_following has invalid value
+                {"ccw_following": "BADVALUE"},
+            ):
+                with self.subTest(bad_config=bad_config):
+                    with self.assertRaises(salobj.ExpectedError):
+                        await self.configure_script(**bad_config)
+
+    async def test_configure_good(self):
+        """Test configure method with valid configurations."""
+        async with self.make_script():
+
+            # Start with default configuration
+            await self.configure_script()
+
+            # The default value for `ccw_following` is True
+            self.assertEqual(self.script.config.ccw_following, True)
+
+            # Now try to change that to False
+            ccw_following = False
+            await self.configure_script(
+                ccw_following=ccw_following,
+            )
+
+            # Check configuration was correctly loaded
+            self.assertEqual(self.script.config.ccw_following, ccw_following)
+
+    async def test_executable(self):
+        scripts_dir = get_scripts_dir()
+        script_path = scripts_dir / "maintel" / "setup_mtcs.py"
+        await self.check_executable(script_path)
+
+    async def test_run(self):
+
+        # Start the test itself
+        async with self.make_script():
+
+            # ToDo - This is not implemented in the mock.
+            #  Emulating here for now.
+            class SimulationMode:
+                mode = 0
+                private_sndStamp = utils.current_tai()
+
+            class SoftwareVersion:
+                cscVersion = "v0.0.0"
+                private_sndStamp = utils.current_tai()
+
+            async def emu_get_simulation_mode(components):
+                """Emulate `mtcs.get_simulation_mode` which is not
+                implemented in MTCSMock."""
+                return {c: SimulationMode for c in components}
+
+            async def emu_get_software_version(components):
+                """Emulate `mtcs.get_software_versions` which is not
+                implemented in MTCSMock"""
+                return {c: SoftwareVersion for c in components}
+
+            self.script.mtcs.get_simulation_mode = emu_get_simulation_mode
```

Comment:
See my comment above about using async mock instead

Reply:
Same here. We do not need these mocks anymore.